rinja_derive/src/generator.rs

```diff
@@ -39,6 +39,8 @@ pub(crate) struct Generator<'a> {
     buf_writable: WritableBuffer<'a>,
     // Counter for write! hash named arguments
     named: usize,
+    // Used in blocks to check if we are inside a filter block.
+    is_in_filter_block: bool,
```

Comment:
You can use `is_in_filter_block: u32`, and use `+= 1` and `-= 1` so you don't need the `prev` variable.